Course IFSC 7399

Assignment 1

Assignment submitted to - Prof NingNing WU

Assignment Submitted by - Deepak Singla



---



Python Lecture 1

Ex7: School Desks

A school decided to replace the desks in three classrooms. Each desk seats two students.

Given the number of students in each class, determine the smallest possible number of desks that can be purchased.

The program should:

Read three integers: the number of students in each of the three classerooms A, B and C respectively.

Use floor division to find the number of desk that has two students and use modulus to find the number of desks that have one student.

Add the number of desks needed for each classroom and print the total.

Example:
Classroom A has 20 students and thus needs 10 desks.

Classroom B has 21 students, so they can get by with no fewer than 11 desks.

Classroom C has 22 and also needs 11 desks

32 desks in total (10 + 11 + 11)

Enter Classroom A: 20

Enter Classroom B: 2

Enter Classroom C: 20

31

Enter Classroom A: 20

Enter Classroom B: 21

Enter Classroom C: 22

32

Enter Classroom A: 21

Enter Classroom B: 22

Enter Classroom C: 23

34



In [4]:
#user to enter the number of students in each classroom
studentsClassroomA = int(input("Enter Classroom A: "))
studentsClassroomB = int(input("Enter Classroom B: "))
studentsClassroomC = int(input("Enter Classroom C: "))

#Calculate the number of desks needed for each classroom
desksClassroomA = studentsClassroomA // 2 + studentsClassroomA % 2
desksClassroomB = studentsClassroomB // 2 + studentsClassroomB % 2
desksClassroomC = studentsClassroomC // 2 + studentsClassroomC % 2

#desks required in class A
print("Desks required in class A : " + str(desksClassroomA))

#desks required  in class B
print("Desks required in class B : " + str(desksClassroomB))

#desks required in class C
print("Desks required in class C : " + str(desksClassroomC))

#Calculate the total number of desks needed
totalDesksRequired = desksClassroomA + desksClassroomB + desksClassroomC


print("Total desks required in school: " + str(totalDesksRequired))




Enter Classroom A: 21
Enter Classroom B: 22
Enter Classroom C: 23
Desks required in class A : 11
Desks required in class B : 11
Desks required in class C : 12
Total desks required : 34


Python Lecture 1

Ex9: Miles in Kilometer

Write a program that prompts for the number of kilometers.

Calculate and display the number of miles in the kilometers.

Hint: Use the float() function to convert your input to an floating number.

Hint: There are 1.61 kilometers in a mile.

Enter Kilometers: 5

3.105590062111801

Enter Kilometers: 0

0.0

Enter Kilometers: 22.5

13.975155279503104

In [5]:
#user to enter number of kilomteres
numOfKilometers = input("Enter Kilometers: ")

print("Kilometers : " + str(numOfKilometers))

#convert the kilometers to float value from string input
numOfKilometers = float(numOfKilometers)

#calculate the number of miles
numOfMiles = numOfKilometers / 1.61

#print the number of miles
print("Miles : " + str(numOfMiles))

Enter Kilometers: 22.5
Kilometers : 22.5
Miles : 13.975155279503104


Python Lecture 2

Ex6: Sum of Digits

Enter a three digit number and print the sum of the digits.

Be sure to use the .format method to create your output.

Use only math for your solution; do no use any string functions.

Enter a 3 Digit Number: 179

Sum of Digits: 17


Enter a 3 Digit Number: 829

Sum of Digits: 19


Enter a 3 Digit Number: 204

Sum of Digits: 6

In [7]:
#user to enter three digit number
numberEntered = input("Enter a three digit number: ")

#if user entered three digit number
print("Three digit number entered : " + str(numberEntered))

#convert string input to int value
numberEntered = int(numberEntered)

#find individual numbers
numberatOnesPlace = numberEntered % 10
numberatTensPlace = (numberEntered // 10) % 10
numberatHundredsPlace = numberEntered // 100

#print all three numbers separately
print("Number at ones place: {}".format(str(numberatOnesPlace)))
print("Number at tens place: {}".format(str(numberatTensPlace)))
print("Number at hundreds place: {}".format(str(numberatHundredsPlace)))

#calc the sum of all three numbers
sumOfDigits = numberatOnesPlace + numberatTensPlace + numberatHundredsPlace

#print sum of all numbers
print("Sum of three Digits: {}".format(str(sumOfDigits)))




Enter a three digit number: 123
Three digit number entered : 123
Number at ones place: 3
Number at tens place: 2
Number at hundreds place: 1
Sum of three Digits: 6


Python Lecture 2

Ex7: Reverse Three Digits

Enter a three digit number and reverse the digits.

Be sure to use to .format method to create your output.

Use only math for your solution; do no use any string functions.

Enter a 3 Digit Number: 291

Reverse of Digits: 192


Enter a 3 Digit Number: 321

Reverse of Digits: 123


Enter a 3 Digit Number: 444

Reverse of Digits: 444

In [9]:
#user to enter three digit number
numberEntered = input("Enter a three digit number: ")

#print user entered three digit number
print("Three digit number entered : " + str(numberEntered))

#convert string input to int value
numberEntered = int(numberEntered)

#find individual numbers
numberatOnesPlace = numberEntered % 10
numberatTensPlace = (numberEntered // 10) % 10
numberatHundredsPlace = numberEntered // 100

print("Number at ones place: {}".format(str(numberatOnesPlace)))
print("Number at tens place: {}".format(str(numberatTensPlace)))
print("Number at hundreds place: {}".format(str(numberatHundredsPlace)))

#concatenate numbers together
reversedNumber = str(numberatOnesPlace) + str(numberatTensPlace) + str(numberatHundredsPlace)

print("Reverse of three digit number: {}".format(reversedNumber))



Enter a three digit number: 123
Three digit number entered : 123
Number at ones place: 3
Number at tens place: 2
Number at hundreds place: 1
Reverse of Digits: 321
